test/rules/consecutive-function-overloads/test.ts.lint

```diff
@@ -0,0 +1,63 @@
+// good
+
+interface i1 {
+    a();
+    a(x: number);
+    b();
+    b(x: string);
+}
+
+interface i2 {
+    a();
+    a(x: number);
+    a(): void;
+    b();
+    b(x: string);
+}
+
+interface i3 {
+    a();
+    "a"();
+}
+
+interface i4 {
+    a();
+    ["a"]();
+}
+
+
+// bad
```

Comment:
yep, I think you'll want to recurse and create a new block scope for the nested member